lib/fluent/plugin/filter_kubernetes_metadata.rb

```diff
@@ -59,55 +62,11 @@ class KubernetesMetadataFilter < Fluent::Filter
 
     config_param :annotation_match, :array, default: []
     config_param :stats_interval, :integer, default: 10
+    config_param :namespace_watch_strategy, :string, default: 'default'
```

Comment:
Can we call this a "cache" strategy instead of a watch strategy?  Or is this really about watch strategies?  If the are going to call this a catch strategy, then the methods of the cache needs to be hidden so that they can be easily replaced.  The API calls are not covered here, right?